src/python/m5/util/convert.py

```diff
@@ -118,6 +133,32 @@ def _split_suffix(value, suffixes):
     return (value[: -len(matches[0])], matches[0]) if matches else (value, "")
 
 
+def _split_suffix_by_prefix(value):
+    """Split a string based on a suffix from a list of suffixes.
+    This is intended to be used with the keys of binary_prefixes.
+
+    :param value: String value to test for a matching magnitude
+    (e.g. k, M, Gi, etc).
+    :param suffixes: Container of suffixes to test.
+
+    :returns: A tuple of (value, suffix). Suffix is the empty string
+              if there is no match.
+
+    """
+    matches = [sfx for sfx in binary_prefixes.keys() if sfx in value]
+
+    if len(matches) == 2:  # e.g. matches both M and Mi
+        matches.sort(key=len, reverse=True)
+        matches.pop()
+        assert len(matches) <= 1
```

Comment:
This is a weird assert for two reasons:

1. If `len(matches) == 2` and we pop once then `matches` is always equal to `1`. I can't see anything here that'll ever result in this assert failing.
2. Why `<=` and not just `==`? This login implies there might be a state where `matches == 0` and, in this case, we want to continue in which case the `return` function will fail when it tries to access `matches[0'`.

Reply:
Would it be best just to remove it? It was carried over from _split_suffix, so it looks like it's useful in _split_suffix but not here.

Reply:
Remove it or change it to `assert len(matches) == 1`. The latter is a pretty meaningless assert but can do no real harm so I'd go with that.